agent_shared/src/config/mod.rs

```diff
@@ -435,8 +435,9 @@ pub struct EventStoreConfig {
 #[serde(rename_all = "snake_case")]
```

Comment:
Could we serde individually? Using `mongodb` instead of `mongo_db` is more conventional. `in_memory` can stay as is.